### DISCUSSION TITLE
Unexpected Git error on GitHub Actions: RPC failed; HTTP 408 curl 18 HTTP/2

### DISCUSSION THREAD

**Discussion by jeanjerome** [community/community#55820] [Actions] [answered]
### Select Topic Area

Bug

### Body

Hello everyone,

Since yesterday, GitHub Actions have been returning errors of the following type when deploying to GitHub Pages:

```
Deploying to scalastic/scalastic.github.io on branch gh-pages
Deploying to ***github.com/scalastic/scalastic.github.io.git
...
Initialized empty Git repository in /github/workspace/build/.git/
[master (root-commit) 4089974] jekyll build from Action c4440fef66d42f3d7031f4ca87911d4a24c14532
 1205 files changed, 25677 insertions(+)
 create mode 100644 .nojekyll
 create mode 100644 404.html
 create mode 100644 CNAME
...
error: RPC failed; HTTP 408 curl 18 HTTP/2 stream 7 was reset
send-pack: unexpected disconnect while reading sideband packet
fatal: the remote end hung up unexpectedly
```

Is there a problem or a change in the GitHub Pages deployment process?
Thank you for your response.

**Comment by jeanjerome** [comment]
I you want to have a look to the complete logs: https://github.com/scalastic/scalastic.github.io/actions/runs/5008728644/jobs/8976906598

**Reply by jeanjerome** [reply]
I tried to install Git LFS on the Action Runner but without any change in the error message.

**Reply by jeanjerome** [reply]
Hey everyone,

I managed to find a temporary solution. Here's what I did:

1. Set the global Git configuration to use HTTP/1.1:
```
git config --global http.version HTTP/1.1
```
This command ensures that Git uses the HTTP/1.1 protocol for network communication.

2. Increase the HTTP post buffer size in the global Git configuration (**Not needed in my case but read the note!**):
```
git config --global http.postBuffer 157286400
```
By setting the post buffer size to 157286400 bytes, we prevent any issues related to large HTTP requests.

> **Note**: *There's some discussion online suggesting that increasing this setting beyond the recommended value might be unnecessary or even risky. Please try it if the fix does not work without this command and exercise caution while using this approach.*

3. Install Git Large File Storage (LFS) (depending on the Linux distribution you are using when the Git commit is done):
- For Ubuntu or Debian-based distributions:
```
apt-get install -y git-lfs
```

- For Fedora:
```
dnf install -y git-lfs
```

- For CentOS or RHEL:
```
yum install -y git-lfs
```

- For Arch Linux:
```
pacman -S --noconfirm git-lfs
```


This command installs Git LFS, which enables the handling of large files efficiently.

These steps provided a temporary workaround for the bug we encountered. Please note that this solution might not address the root cause of the issue, but it should allow us to continue working without interruptions. Make sure to revert these changes once an official solution becomes available.

Don't hesitate to provide feedback on the bug resolution.

**Reply by BryanSchuetz** [reply]
Where in your workflow do you do this?

**Reply by mkjsix** [reply]
In my case I think this would not be effective, as the issue happens during the workflow itself: it seems it's an internal error when the `helaili/jekyll-action@v2` tries to update the gh-pages branch, so that is not under our control.

![Screenshot 2023-05-19 alle 11 55 30](https://github.com/community/community/assets/365382/2f0e66db-4ff3-411e-9a05-c6d6f353bed5)

**Reply by alexanderameye** [reply]
> In my case I think this would not be effective, as the issue happens during the workflow itself: it seems it's an internal error when the `helaili/jekyll-action@v2` tries to update the gh-pages branch, so that is not under our control.
> 
> ![Screenshot 2023-05-19 alle 11 55 30](https://github.com/community/community/assets/365382/2f0e66db-4ff3-411e-9a05-c6d6f353bed5)
> 

Same for me, although I am using a different action 

https://github.com/cpina/github-action-push-to-another-repository

But it's the same thing essentially I think

**Reply by jeanjerome** [reply]
@BryanSchuetz 
I'm using this Actions: [jeffreytse/jekyll-deploy-action](https://github.com/jeffreytse/jekyll-deploy-action)

You can have a look at my own workflow here with the added commands: https://github.com/scalastic/scalastic.github.io/blob/fec3121610817756bea3859f610bf986587e8aaa/.github/workflows/build-jekyll.yml#L37

**Reply by jeanjerome** [reply]
@mkjsix 
For `helaili/jekyll-action`, you have the [pre_build_commands](https://github.com/helaili/jekyll-action#pre_build_commands) inputs. Just set it like:
 ```
git config --global http.version HTTP/1.1; apk fetch -y git-lfs;
```
(not sure about the `-y` but you get the idea)

**Reply by EverettBerry** [reply]
Thank you guys very much for this solution. This is the exact line I added to helaili/jekyll-action@v2 workflow and it worked. 

```
with:
    pre_build_commands: git config --global http.version HTTP/1.1; apk fetch git-lfs;
```

**Reply by mkjsix** [reply]
Thank you @jeanjerome @EverettBerry, your solution is working fine for me.

**Reply by aminvakil** [reply]
AFAICS these are workarounds, not permanent solution. This should not be marked as accepted answer.

**Reply by jeanjerome** [reply]
@aminvakil You are correct, this is not a definitive answer as it has been stated, but rather an actual solution for implementation using GitHub Actions.

**Reply by jeffreytse** [reply]
@jeanjerome Now this issue has been fixed for using this action [jeffreytse/jekyll-deploy-action](https://github.com/jeffreytse/jekyll-deploy-action)

**Reply by jeanjerome** [reply]
@jeffbeehler Thank you for your solution. I noticed that you used `http.postBuffer` instead of `git-lfs`. Is there a specific reason for this choice? I will conduct further tests.

**Reply by jeffreytse** [reply]
@jeanjerome Hi, it's a minimum solution without bringing in other dependencies and drawbacks.

**Reply by Jason2866** [reply]
It is not a fix! It is a workaround. Thx for the workaround. 
BUT the issue hits a lot of GH Actions and it is not the solution to add workarounds everywhere.
Github needs to find the reason and fix it or tell officialy what changes are needed.

**Reply by jeffreytse** [reply]
@Jason2866 You're right, but before the final official solution coming out, it's a good workaround so call "fix" to settle the issue effectively.

**Reply by nihaals** [reply]
I think the rationale for not wanting it to be marked as the answer is so GitHub is more likely to see this discussion as opposed to assuming it is "solved" so it is investigated sooner (assuming they haven't investigated based on support tickets and possible server logs).

**Reply by aminvakil** [reply]
@nihaals Thanks, exactly!

**Reply by Sanni-Damilola** [reply]
jobs:
  your_job:
    runs-on: ubuntu-latest
    env:
      GIT_HTTP_TIMEOUT: 300  # Set a higher timeout value (e.g., 5 minutes)
    steps:
      - name: Your Git action step
        # Your Git-related actions here

**Comment by mkjsix** [comment]
We have the same error here:
https://github.com/SoftInstigate/restheart-website/actions/runs/5013979152/jobs/8987705668 

The error log ends with:

```
...
 create mode 100644 use-cases/rest-api.html
error: RPC failed; HTTP 408 curl 18 HTTP/2 stream 7 was reset
send-pack: unexpected disconnect while reading sideband packet
fatal: the remote end hung up unexpectedly
Everything up-to-date
```

The workflow is:
https://github.com/SoftInstigate/restheart-website/blob/master/.github/workflows/jekyll.yml

**Comment by aminvakil** [comment]
Same here:
https://github.com/aminvakil/aur/actions/runs/5009690513/jobs/8988928436#step:11:177
error: RPC failed; HTTP 408 curl 18 HTTP/2 stream 7 was reset

**Comment by BryanSchuetz** [comment]
Having the same issue

**Comment by JunchengXue** [comment]
Having the same issue

**Comment by BryanSchuetz** [comment]
Have tried a number of things to get around this issue but nothing seems to work. I though maybe the repo was to large, but reducing the size and trying again hasn't had an impact. This action has been working many times a day for months with no issue, until yesterday... not sure how to address short of just completely rewriting the workflow, but even then I'm not sure the same thing still won't happen.

**Reply by EverettBerry** [reply]
We are having the same issue. Reduced about 50 MB from the app to no effect.

**Comment by dharmenbavaria** [comment]
Having the same issue, changes to one specific file fails to push everything with the same error mentioned in the main discussion, if you revert that file change it git push works, so there is definitely something going wrong.

**Comment by alexanderameye** [comment]
Exact same issue since recently +1

**Comment by mkjsix** [comment]
In our case, the workflow has been running for months without any issue and the last successful execution was last week on May 12th 2023. So, I think the problem can't be the content of our repository, which has not changed significantly over the last year, but definitely some recent modification of the execution environment of the GitHub actions. 
Yesterday I also opened a ticket with Github support, but I have not received any feedback so far.

**Comment by Jason2866** [comment]
Same issue recently +1
workflow did run for month.

EDIT:
Adding this in my used forked action https://github.com/Jason2866/copy_file_to_another_repo_action/tree/http408_fix
```
git config --global http.version HTTP/1.1
git config --global http.postBuffer 157286400
```
like @jeanjerome did worked. Solved. Thx! (git lfs is already installed)

**Comment by aminvakil** [comment]
Is github going to be this way or is this a bug? Is there an official response regarding this mater?

**Reply by mkjsix** [reply]
@aminvakil I opened a ticket and GitHub support suggested having a look at https://github.com/helaili/jekyll-action/issues/143, because this issue is related to a third-party action that it's not officially supported.

So, the solution to fix the `helaili/jekyll-action@v2` seems to be [what has been already suggested above](https://github.com/orgs/community/discussions/55820#discussioncomment-5946136). In my case, it works if I add the pre_build_commands:

```
    - uses:  helaili/jekyll-action@v2
      with:
        pre_build_commands: git config --global http.version HTTP/1.1; apk fetch git-lfs;
        token: ${{ secrets.GITHUB_TOKEN }}
        target_branch: 'gh-pages'
```

**Reply by aminvakil** [reply]
I'm not using `helaili/jekyll-action@v2`.
https://github.com/aminvakil/aur/actions/runs/5029785702/jobs/9021743297#step:11:177
It's basically doing `git clone https://github.com/ansible/ansible` in GHA.

**Reply by d-otis** [reply]
Thank you @mkjsix for specifying where to place the `git config` change.
Works like a charm! 🤩 🙏🏻 👏🏻

**Reply by julian-poidevin** [reply]
To answer @aminvakil : should an issue be opened on GitHub ? I doubt this discussion will have enough visibility.

**Reply by aminvakil** [reply]
@julian-poidevin I don't know how github works, whether they prioritize an issue or a discussion or something else.

As this discussion is already in home page of https://support.github.com/features/actions on "Helpful community posts", I'm not sure how to drag attention of someone from github here to this problem.

All I'm saying is if I had a list of issues to take care of (as a person, team, organization, etc.) I would focus more on issues that don't have a "Accepted answer".

I may be totally wrong and something else is going on in github though.

**Comment by MarkParker5** [comment]
Same issue since recently, waiting for an official response.

**Comment by davidwarner2212** [comment]
I have facing the same problem and just waiting for more good [responses ](https://dirtbikecoach.com/)to resolve it appropriately.
Thanks for information[.](https://dirtbikecoach.com/)

**Comment by aminvakil** [comment]
I just got this error on my system as well, seems it's not related just to GHA.
```
$ git --version
git version 2.40.1
```

**Comment by CometVoid** [comment]
I made a brand new repo specifically to try avoid this frustrating bug that seemingly is only effecting some repos. Github let me make 1 commit, but no more after that.

I just want to backup my notes man, but 'git push' no worky ;-;

**Comment by dobbelina** [comment]
Thanks for the fix @Jason2866
I have incorporated it into my own fork.

**Comment by Foxboron** [comment]
The same issue is preventing packages from being packaged on @ArchLinux.

`strace git clone`  stalls at `write(4, "0011command=fetch0014agent=git/2"..., 1443702) = 1443702` before getting the `408`.

```
  -> Cloning moby git repo...
Cloning into bare repository '/home/foxboron/.cache/offload-build/tmp.6JYKszWSrz/moby'...
error: RPC failed; HTTP 408 curl 18 HTTP/2 stream 7 was reset
fatal: expected 'packfile'
```

No update from the Github team on this?

**Reply by aksuhton** [reply]
Also on arch and experienced an issue pushing to one of my repositories today that I was able to push to two days ago.

error: RPC failed; HTTP 408 curl 18 HTTP/2 stream 7 was reset
send-pack: unexpected disconnect while reading sideband packet
fatal: the remote end hung up unexpectedly

**Reply by Foxboron** [reply]
Seems like I can reproduce it by trying to clone deno over https as reported by @felixonmars.

```
λ ~ » git clone https://github.com/denoland/deno.git
Cloning into 'deno'...
remote: Enumerating objects: 110410, done.
remote: Counting objects: 100% (26165/26165), done.
remote: Compressing objects: 100% (1914/1914), done.
remote: Total 110410 (delta 24759), reused 24427 (delta 24241), pack-reused 84245
Receiving objects: 100% (110410/110410), 99.29 MiB | 7.72 MiB/s, done.
Resolving deltas: 100% (84310/84310), done.
λ ~ » cd deno
λ deno main Ɇ »
λ deno main Ɇ » git submodule update --init
Submodule 'test_util/std' (https://github.com/denoland/deno_std) registered for path 'test_util/std'
Submodule 'test_util/wpt' (https://github.com/web-platform-tests/wpt.git) registered for path 'test_util/wpt'
Submodule 'deno_third_party' (https://github.com/denoland/deno_third_party.git) registered for path 'third_party'
Submodule 'tools/node_compat/node' (https://github.com/denoland/node_test.git) registered for path 'tools/node_compat/node'
Cloning into '/home/fox/deno/test_util/std'...
Cloning into '/home/fox/deno/test_util/wpt'...
error: RPC failed; HTTP 408 curl 18 HTTP/2 stream 7 was reset
fatal: expected 'packfile'
```

Generally this seems to be a larger issue with the Github git HTTP rpc server?

**Reply by julian-poidevin** [reply]
I really think you should be opening an issue if you want this to get resolved.
This discussion has been marked "Resolved" by the original author so I don't think GitHub will look at it 🫤.

**Reply by dobbelina** [reply]
> I really think you should be opening an issue if you want this to get resolved. This discussion has been marker "Resolved" by the original author so I don't think GitHub will look at it 🫤.

Good point.
@jeanjerome should remove the "Answered" label as the given information is not a fix but a workaround.

**Reply by jeanjerome** [reply]
Alright guys, I'm removing the "resolved" label and let's see if it unlocks the situation.

**Reply by tpkessler** [reply, marked as answer]
The testing repository of Arch Linux contains an updated `curl-8.1.1-2` that includes https://github.com/curl/curl/commit/5c58cb0212bcf63cce33a974906bf9905948b4bb. This fixes the problem for me.

**Reply by jeanjerome** [reply]
Good catch @tpkessler

- This could be an issue with cURL, which in version 8.1.0 introduced support for HTTP/2 ( search [CURLPROXY_HTTPS2: for HTTPS proxy that may speak HTTP/2](https://curl.se/changes.html#8_1_0)), and version 8.1.1 fixed an issue with uploads (in [http2: upload improvements](https://curl.se/changes.html#8_1_1)).
- Additionally, the release date of version 8.1.0 aligns with the occurrence of the bug: May 17, 2023.

To be tested as soon as possible...

**Reply by aminvakil** [reply]
My issue has been resolved too with curl 8.1.1.

**Reply by jeanjerome** [reply]
Yes, it's working well now! Thanks.

**Reply by d-otis** [reply]
Working for me as well. TY for posting the answer! 🙏🏻

**Comment by langroodi** [comment]
From yesterday the issue is disappeared from my [GitHub action](https://github.com/marketplace/actions/doxygenize) and now it is working as normal. Are you experiencing also the normal situation again or the `git` on the GitHub runners works occasionally (sometimes good and sometimes bad)?

**Comment by dobbelina** [comment]
I assume this means we can remove the below ~fix~ work-around then?
```
git config --global http.version HTTP/1.1
git config --global http.postBuffer 157286400
```

**Reply by jeanjerome** [reply]
@dobbelina Yes, there is no longer a need for it

**Reply by cmxl** [reply]
I still had to set the http version to 1.1 although using curl 8.1.2 on Windows 10.
Git version 2.41.0 for Windows. 
But i had to push changes to Azure DevOps.
Is there someone else with this problem?

**Reply by julian-poidevin** [reply]
The issue just re-surfaced for me, it was fixed before.
